smartsim/entity/dbobject.py

```diff
@@ -107,18 +109,24 @@ def _enumerate_devices(self) -> t.List[str]:
         :rtype: list[str]
         """
         devices = []
-        if ":" in self.device and self.devices_per_node > 1:
-            msg = "Cannot set devices_per_node>1 if a device numeral is specified, "
-            msg += f"the device was set to {self.device} and devices_per_node=={self.devices_per_node}"
-            raise ValueError(msg)
-        if self.device in ["CPU", "GPU"] and self.devices_per_node > 1:
+        if self.device in ["GPU"] and self.devices_per_node > 1:
```

Comment:
Just check against GPU (no `in` needed). I suppose too we could always make CPU and GPU something like an `enum` and attach them to `DBObject` so that we're comparing an object of some variety instead of a magic string